src/security/decorators.py

```diff
@@ -952,3 +953,29 @@ def deny_access(request, *args, **kwargs):
     )
 
     raise PermissionDenied(*args, **kwargs)
+
+
+def article_stage_review_required(func):
+    """
+    Checks that the article is in one of the review stages
+    :param func: func
+    :return: PermissionDenied or func
+    """
+
+    def review_required_wrapper(request, *args, **kwargs):
+        article_id = kwargs.get('article_id', None)
```

Comment:
This means that the article_id MUST be a kwarg.
If that is the case, python3 introduced a cool feature to express this in the function signature:

```
def review_required_wrapper(request, *args, article_id=None,**kwargs):
     if article_id is None:
          raise Http404
     [...]
```

Reply:
Further changes made.